cmd/clusterctl/client/repository/components.go

```diff
@@ -292,9 +301,73 @@ func addNamespaceIfMissing(objs []unstructured.Unstructured, targetNamespace str
 	return objs
 }
 
+func fixWebHookTargetNamespace(o unstructured.Unstructured, targetNamespace string) (unstructured.Unstructured, error) {
```

Comment:
Can you move this func below fixTargetNamespace? (just easier to read top-down imho)

Reply:
I think we should rename the func because it also fixes CRDs

Reply:
sure, can move it. My idea behind the name was it fixes the webhook references in the crd.. but if you have a name i'll happily change it.

Reply:
@asalkeld

Ah, of course, wasn't thinking about that. The func also updates the injected secret references not only the webhooks. 

One idea I would have is `fixTargetNamespaceClusterWideResources` (so this includes CRDS, WebhookConfigurations and all the fields we have to patch inside them: secret ref + webhooks)

Reply:
Ah let's keep it. I think yours is good. Mine is not entirely correct because it only fixes CRDs/Webhooks. Let's just count the secret ref overall as a part of a webhook (which it is of course because we need the secret for the webhook) and we're good ;)